src/SwedbankPay.Sdk.Tests/Json/UknownEnumTests.cs

```diff
@@ -1,43 +1,45 @@
-        [InlineData(typeof(PriceType), "Bottle")]
-        [InlineData(typeof(TransactionType), "Notebook")]
-        [InlineData(typeof(Intent), "Eraser")]
-        public void DeserializingUnknownEnum_GivesDefaultValue(Type enumType, string value)
-        {
-            JsonConvert.DeserializeObject($"\"{value}\"", enumType, JsonSerialization.JsonSerialization.Settings);
-        }
-
-        [Fact]
-        public void DeserializingUnknownEnum_DoesNotThrowForAnyType()
-        {
-            var enums = Assembly.GetAssembly(typeof(PaymentInstrument)).GetTypes().Where(t => t.IsEnum && !t.IsDefined(typeof(FlagsAttribute)));
-            foreach (var item in enums)
-            {
-                var result = JsonConvert.DeserializeObject(InvalidStringEnum, item, JsonSerialization.JsonSerialization.Settings);
-                Assert.True(result.ToString().Equals("Unknown", StringComparison.InvariantCultureIgnoreCase), $"The default value for {item.Name} is not the string \"Unknown\"");
-            }
-        }
-
-        public static string InvalidStringEnum = "\"ThisIsAVeryInvalidEnumValueForThisObject\"";
-    }
-}
+﻿using SwedbankPay.Sdk.Payments;
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+
+using Xunit;
+
+namespace SwedbankPay.Sdk.Tests.Json
+{
+    public class UknownEnumTests
+    {
+        [Fact]
+        public void DeserializingAUnknownEnum_DoesNotThrowAnError()
+        {
+            var paymentInstrument = JsonSerializer.Deserialize<PaymentInstrument>(InvalidStringEnum, JsonSerialization.JsonSerialization.Settings);
+            Assert.Equal(PaymentInstrument.Unknown, paymentInstrument);
+        }
+
+        [Theory]
+        [InlineData(typeof(PaymentInstrument), "Chair")]
+        [InlineData(typeof(PaymentInstrument), "Bottle")]
+        [InlineData(typeof(PriceType), "Bottle")]
+        [InlineData(typeof(TransactionType), "Notebook")]
+        [InlineData(typeof(Intent), "Eraser")]
+        public void DeserializingUnknownEnum_GivesDefaultValue(Type enumType, string value)
+        {
+            JsonSerializer.Deserialize($"{{ \"propertyName\":\"{value}\" }}", enumType, JsonSerialization.JsonSerialization.Settings);
+        }
+
+        [Fact]
+        public void DeserializingUnknownEnum_DoesNotThrowForAnyType()
+        {
+            var enums = Assembly.GetAssembly(typeof(PaymentInstrument)).GetTypes().Where(t => t.IsEnum && !t.IsDefined(typeof(FlagsAttribute)));
+            foreach (var item in enums)
+            {
+                var result = JsonSerializer.Deserialize(InvalidStringEnum, item, JsonSerialization.JsonSerialization.Settings);
+                Assert.True(result.ToString().Equals("Unknown", StringComparison.InvariantCultureIgnoreCase), $"The default value for {item.Name} is not the string \"Unknown\"");
+            }
+        }
+
+        public static string InvalidStringEnum = $"{{ \"propertyName\":\"ThisIsAVeryInvalidEnumValueForThisObject\"}}";

```

Comment:
Codacy found an issue: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)